dynamite_flask_app/create_update_database.py

```diff
@@ -0,0 +1,188 @@
+    },
+    {
+        'name': 'Ahmed',
+        'email': 'abhhewc1@123.com',
+        'password': 'asd123',
+        'address': 'ab2erc',
+        'contact_no': 176474523,
+    },
+    {
+        'name': 'Boy',
+        'email': 'furrfurr@123.com',
+        'password': 'asd123',
+        'address': 'fwefeew',
+        'contact_no': 8484876,
+    }
+]
+
+orders = [
+    {
+        'date': '2008, 4, 12'
+    },
+    {
+        'date': '2018, 12, 02'
+    }
+]
+
+
+def add_products():
+    """Adds all products with their details to the database."""
+    for product in products:
+        product_details = Product(
+            product_id=product['id'], name=product['name'], price=product['price'], sale_price=product['sale_price'],
+            description=product['description']
+        )
+
+        add_image_url(product, product_details)
+        add_feature(product, product_details)
+        add_category(product_details)
+        add_size(product, product_details)
+
+        db.session.add(product_details)
+
+    db.session.commit()
+
+
+def add_image_url(product, product_details):
+    """Adds image urls to the database.
+
+    Args:
+        Product(str): Each product, used from the add_products() method.
+        product_details: Add product to the database, also used to create relationship with the Product table.
+
+    """
+    for image_url in product['img_urls']:
+        url = ImageUrl(url=image_url)
+        product_details.img_urls.append(url)
+
+
+def add_feature(product, product_details):
+    """adds features to the database.
+
+    Args:
+        Product(str): Each product, used from the add_products() method.
+        product_details: Add product to the database, also used to create relationship with the Product table.
+
+    """
+    for detail in product['details']:
+        feature = Feature.query.filter_by(feature=detail).first()
+        if feature is None:
+            feature = Feature(feature=detail)
+        product_details.features.append(feature)
+
+
+def add_size(product, product_details):
+    """adds sizes to the database.
+
+    Args:
+        Product(str): Each product, used from the add_products() method.
+        product_details: Add product to the database, also used to create relationship with the Product table.
+
+    """
+    for each_size in product['sizes']:
+        each_size = each_size.upper()
+        available = ProductSize(availability=bool(random.getrandbits(1)))
+
+        size = Size.query.filter_by(size=each_size).first()
+
+        if size is None:
+            size = Size(size=each_size)
+
+        available.size = size
+        available.product = product_details
+
+
+def add_category(product_details):
+    """adds categories to the database.
+
+    Args:
+        product_details: Add product to the database, also used to create relationship with the Product table.
+
+    """
+    category = Category.query.filter_by(type='General').first()
+    if category is None:
+        category = Category()
```

Comment:
adding categories without any type.

Reply:
This is because I've set the category's default to general, I'll change it here if you say so.